src/Land/Model/soil_heat.jl

```diff
@@ -1,35 +1,251 @@
 ### Soil heat model
+
 export SoilHeatModel, PrescribedTemperatureModel
 
 abstract type AbstractHeatModel <: AbstractSoilComponentModel end
 
 """
-    PrescribedTemperatureModel{FT} <: AbstractHeatModel
+    struct PrescribedTemperatureModel{F1} <: AbstractHeatModel
 
-A model where the temperature is set by the user and not dynamically determined.
+Model structure for a prescribed temperature model.
+
+Document.
 
-This is useful for driving Richard's equation without a back reaction on temperature.
 # Fields
 $(DocStringExtensions.FIELDS)
 """
-Base.@kwdef struct PrescribedTemperatureModel{FT} <: AbstractHeatModel
-    "Prescribed function for temperature"
-    T::FT = FT(0.0)
+struct PrescribedTemperatureModel{F1} <: AbstractHeatModel
+    "Temperature"
+    T::F1
 end
 
 """
-    get_temperature(m::PrescribedTemperatureModel)
+    PrescribedTemperatureModel(
+        T::Function = (aux,t) -> eltype(aux)(0.0)
+    )
+
+Outer constructor for the PrescribedTemperatureModel defining default values, and
+making it so changes to those defaults are supplied via keyword args.
```

Comment:
same as for PrescribedWater - we arent using kwargs anymore. "Outer constructor for the PrescribedTemperatureModel defining default values" is probably enough description

Reply:
done